packages/fuels-core/src/types/wrappers/transaction.rs

```diff
@@ -486,3 +499,36 @@ impl ScriptTransaction {
         self
     }
 }
+
+#[cfg(test)]
+mod test {
+
+    use fuel_tx::policies::Policies;
+
+    use super::*;
+
+    #[test]
+    fn append_witnesses_returns_error_when_limit_exceeded() {
+        let mut tx = ScriptTransaction {
+            tx: FuelTransaction::script(
+                0,
+                vec![],
+                vec![],
+                Policies::default(),
+                vec![],
+                vec![],
+                vec![],
+            ),
+            is_using_predicates: false,
+        };
+
+        let witness = vec![0, 1, 2].into();
+        let err = tx.append_witness(witness).expect_err("should error");
+
+        let expected_err_str = "Validation error: Witness limit exceeded. \
+                                Consider setting the limit manually with a transaction builder. \
```

Comment:
Is `witness_limit ` a mealable field? Could it be adjusted on the transaction level instead of builder level?

Reply:
`witness_limit` is not malleable. So it has to be adjusted on the builder level.